### ISSUE TITLE
Universal zoom

### ISSUE THREAD

**Issue by strowk** [zed-industries/zed#21816]
### Check for existing issues

- [X] Completed

### Describe the feature

I believe it would be more convenient to zoom everything on Ctrl- / Ctrl+ instead of only code editor. 

### Environment

Zed: v0.164.2 (Zed)
OS: Linux X11 ubuntu 24.04
Memory: 60.6 GiB
Architecture: x86_64
GPU: AMD Radeon Graphics (RADV GFX1103_R1) || radv || Mesa 24.0.9-0ubuntu0.2

### If applicable, add mockups / screenshots to help present your vision of the feature

Check how it is done in VS Code would be the simplest

**Comment by notpeter** [comment]
So you'd like it if the UI (tab bar, status bar, etc) zoomed in/out at the same time as the currently zoomable parts of zed (editor, assistant, terminal). This behavior matches VSCode, but differs from most apps: Chrome, JetBrains, Sublime, macOS terminal, etc. 

This would functionality could be implemented either as a setting (e.g. `zoom_everything`) or as an alternate action which could be bound to `ctrl--` and `ctrl-+` in your user keymap.  

As a workaround you can specify a larger font size for one or both of these via settings:
```
  "buffer_font_size": 16,
  "ui_font_size": 16,
```   

See also:

- #5198 
- https://github.com/zed-industries/zed/issues/14613

Thanks for reporting.

**Comment by strowk** [comment]
> This behavior matches VSCode, but differs from most apps: Chrome, JetBrains, Sublime, macOS terminal, etc.

Yes, I also honestly always found that all of those most apps are quite inconvenient to use in regards to zooming. Majority isn't always correct. This is why I have to avoid using anything JetBrains nowadays and browsers typically scale everything except their own small bar that I rarely if at all need to read. 99% of the content is on the web page.

> This would functionality could be implemented either as a setting (e.g. zoom_everything) or as an alternate action which could be bound to ctrl-- and ctrl-+ in your user keymap.

Either sounds good to me :)

> As a workaround you can specify a larger font size for one or both of these via settings
I mostly need zoom because I use a bunch of screens with different DPI and Ubuntu cannot auto-scale properly (at least with X11).
Changing numbers in some file every time I move window around would be a pain. 

I also report this as a suggestion to try to maybe unify zooming somehow, so that any components added in future would zoom OOB without additional requests. This for example includes not only changing font size, but also changing size of some items that contain text (like bars, buttons in them, etc).

**Comment by forinda** [comment]
Any update on this

**Comment by niklaswimmer** [comment]
FWIW, JetBrains IDEs (at least IntelliJ and Android Studio) have an option to zoom the whole IDE, which I have found to be useful in a few occasions already. Leaving some screenshots as inspiration.

<details><summary>Screenshots</summary>
<p>

![Image](https://github.com/user-attachments/assets/5a13cde2-4b5a-410c-a503-1fa75f768104)

![Image](https://github.com/user-attachments/assets/a4ef834c-8859-4c20-867f-6e969bb4659d)

</p>
</details>

**Comment by nilehmann** [comment]
FYI I've been using the following workaround to increase UI and buffer font with a single keystroke

```json
  {
    "bindings": {
      "ctrl-Z u +": "zed::IncreaseUiFontSize",
      "ctrl-Z u -": "zed::DecreaseUiFontSize",
      "ctrl-Z u 0": "zed::ResetUiFontSize",
      "ctrl-Z b +": "zed::IncreaseBufferFontSize",
      "ctrl-Z b -": "zed::DecreaseBufferFontSize",
      "ctrl-Z b 0": "zed::ResetBufferFontSize",
      "ctrl-0": ["workspace::SendKeystrokes", "ctrl-Z u 0 ctrl-Z b 0"],
      "ctrl-=": ["workspace::SendKeystrokes", "ctrl-Z u + ctrl-Z b +"],
      "ctrl-+": ["workspace::SendKeystrokes", "ctrl-Z u + ctrl-Z b +"],
      "ctrl--": ["workspace::SendKeystrokes", "ctrl-Z u - ctrl-Z b -"]
    }
  },
```

**Comment by KhoaLee** [comment]
Thank @nilehmann for the great bindings. It worked with the explorer and the buffer. 

@notpeter I have tried the configuration and tested the trigger from cmd + p to increase font size manually, I noticed that the agent panel (chat panel) does not increase the font size whilst the project explorer did. Could you check on that? I were using the key bindings from the nielhmann.

**Comment by notpeter** [comment]
Yeah, that's managed separately via the `agent_font_size` setting.
At the moment there aren't dedicated actions for changing that. Sorry.

**Comment by mdnjohn** [comment]
I really can't use the editor because of this, the ppi on my monitor is quite dense and i need to zoom in massively to  work, but file bar is just static and i and i can't seems to get  it zoom in.

**Comment by hungryhorace** [comment]
I frequently move between laptop screen and then (with closed lid) plug in a 42" monitor. VS Code or Jetbrains let me easily zoom to taste in a second.

I may be applying @nilehmann 's bindings wrongly. The editor/terminal and the AI panel zoom (whichever is in focus). The explorer never zooms.
